tutorials/pde/2_advection_diffusion.py

```diff
@@ -0,0 +1,178 @@
+"""
+Advection-Diffusion Equation
+============================
+
+Here we use the discretize package to model the advection-diffusion
+equation. The goal of this tutorial is to demonstrate:
+
+    - How to solve time-dependent PDEs
+    - How to apply Neumann boundary conditions
+
+
+Derivation
+----------
+
+If we assume the fluid is incompressible and that diffusivity is spatially,
+the advection-diffusion equation with Neumann boundary conditions is given by:
+
+.. math::
+    p_t = \\alpha \\nabla \\cdot \\nabla p
+    - \\mathbf{u} \\cdot \\nabla p + s \n
+    \\textrm{s.t.} \;\;\; \\frac{\\partial p}{\\partial n} \Bigg |_{\partial \Omega} = 0
+
+where :math:`p` is an unknown variable, :math:`\\alpha` is the diffusivity
+constant, :math:`\\mathbf{u}` the velocity field and :math:`s` is the source
+term. We will consider the case where there are two point sources within our
+domain. Thus:
+
+.. math::
+    s = s_0 \\big [ \\delta ( \\mathbf{r_1}) + \\delta (\\mathbf{r_2} ) \\big ]
+
+To solve this problem numerically, we use the weak formulation; that is, we
+take the inner product of the equation with an appropriate test function
+(:math:`\\psi`):
+
+.. math::
+    \int_\\Omega \\psi \\, p_t \\, dv =
+    \\alpha \int_\\Omega \\psi (\\nabla \\cdot \\nabla p) \\, dv
+    - \int_\\Omega \\psi \\, \mathbf{u} \\cdot \\nabla p \\, dv
+    + \int_\\Omega \\psi \\, s \\, dv
+
+The source term is a volume integral containing the Dirac delta function, thus:
+
+.. math::
+    \int_\\Omega \\psi \\, s \\, dv = \\psi \\, q
+
+where :math:`q=s_0` at the location of either point source and zero everywhere
+else. In the weak formulation, we generally remove divergence terms. If we use
+the identity
+:math:`\\phi \\nabla \\cdot \\mathbf{a} = \\nabla \\cdot (\\phi \\mathbf{a}) - \\mathbf{a} \\cdot (\\nabla \\phi )`
+and apply the divergence theorem we obtain:
+
+.. math::
+    \int_\\Omega \\psi \\, p_t \\, dv =
+    \\alpha \int_{\\partial \\Omega} \\mathbf{n} & \\cdot ( \\psi \\nabla p ) \\, da
+    - \\alpha \int_\\Omega (\\nabla \\psi ) \\cdot (\\nabla p ) \\, dv \n
+    &- \int_{\\partial \\Omega} \\mathbf{n} \\cdot ( \\psi \\, p \\, \\mathbf{u}) \\, da
+    + \int_\\Omega \\big ( \\nabla \\cdot (\mathbf{u} \\, \\psi ) \\big) \\, p \\, dv
+    + \\psi \\, q
+
+Since the flux on the faces is zero at the boundaries, we can eliminate
+the first term on the right-hand side. By defining :math:`\\mathbf{n \\cdot u} = 0`
+on the boundary, the third term on the right-hand side is zero. By evaluating
+the inner products according to the finite volume approach we obtain:
+
+.. math::
+    \\mathbf{\\psi^T M_c p_t} = - \\alpha \\mathbf{\\psi^T G^T M_f G \\, p}
+    + \\mathbf{\\psi^T A_{cf}^T} \\, \\textrm{diag} ( \\mathbf{u} ) \\mathbf{D^T \\, M_c \\, p}
+    + \\mathbf{\\psi^T M_c s}
```

Comment:
I would define `s` as an integrated property rather than introducing `Mc`, 
then you can evaluate analytic integration of delta function, which is your source term.